async_downloads/static/js/ws_async_downloads.js

```diff
@@ -0,0 +1,179 @@
+$(function () {
+    const url = $("#async-downloads-script").data("url");
+    function readablePeriod(timeStamp) {
+        //determine the timezone offset the browser applies to Date()
+        const offset = new Date().getTimezoneOffset() * 60000;
+
+        const diff = Date.now() - Date.parse(timeStamp) + offset;
+        function suffix(number) {
+            return (number > 1 ? "s" : "") + " ago";
+        }
+        let temp = diff / 1000;
+        const years = Math.floor(temp / 31536000);
+        if (years) return years + " year" + suffix(years);
+        const days = Math.floor((temp %= 31536000) / 86400);
+        if (days) return days + " day" + suffix(days);
+        const hours = Math.floor((temp %= 86400) / 3600);
+        if (hours) return hours + " hour" + suffix(hours);
+        const minutes = Math.floor((temp %= 3600) / 60);
+        if (minutes) return minutes + " minute" + suffix(minutes);
+        const seconds = Math.floor(temp % 60);
+        if (seconds) return seconds + " second" + suffix(seconds);
+        return "now";
+    }
+
+    function calculateTimeStamps() {
+        $(".download-timestamp").each(function () {
+            const timeStamp = $(this).attr("value");
+            $(this).text(readablePeriod(timeStamp));
+        });
+    }
+
+    function refreshDownloads() {
+        // Do nothing if the downloads div doesn't exist
+        if ($("#async-downloads").length === 0) return;
+        $.ajax({
+            method: "GET",
+            url: url,
+        }).done(function (response) {
+            $("#async-downloads .dropdown-menu").html(response.html);
+            calculateTimeStamps();
+        });
+    }
+
+    function initDownload(data) {
+        const dropdown = $("#downloads-dropdown");
+        // remove 'No downloads' msg if exists in dropdown
+        dropdown.find("a").remove();
+        const download = data.download;
+        const label = download.app_name
+            ? `<div class="ws-product-label background-color-${download.app_name}"></div>`
```

Comment:
I feel like this is a bit specific. Instead of `app_name`, we could offer a `css_class`, or maybe there's another way to get that information across? Remember we also would like this to work fine for a single app that doesn't necessarily want to style things this way. And we wouldn't want that `ws-product-label` class, unless we can make it something generic.

Reply:
450647a086473ed6014cbfb862f09d913e3af22b